File: src/gestorAplicacion/horario/Dia.java
package gestorAplicacion.horario;

import java.util.ArrayList;
import java.io.Serializable;

import gestorAplicacion.tareas.Tarea;

//En los objetos de esta clase se guardaran las tareas a realizar durante este dia

public class Dia implements Serializable{
	public String etiqueta; //nombre-descripcion que el usuario quiera dar al dia EJ: "Dia de trabajo"
	protected ArrayList<Tarea> tareas; //lista de tareas correspondiente al dia
	private String fecha;
	
	public Dia(String etiqueta, String fecha) {
		this.etiqueta = etiqueta;
		this.tareas = new ArrayList<Tarea>();
		this.fecha = fecha;
	}
	
	
	public String getFecha() {
		return fecha;
	}

	public void setFecha(String fecha) {
		this.fecha = fecha;
	}

	public Dia(String etiqueta, ArrayList<Tarea> tareas, String fecha) { //constructor con etiqueta y tarea
		this.etiqueta = etiqueta;
		this.tareas = tareas;
		this.fecha = fecha;
	}
	
	public Dia(ArrayList<Tarea> tareas, String fecha) { //contructor con etiqueta por defecto
		this("Dia", tareas, fecha);
	}
	
	public void setEtiqueta(String etiqueta) {
		this.etiqueta = etiqueta;
	}
	
	public String getEtiqueta() {
		return etiqueta;
	}
	
	public void setTareas(ArrayList<Tarea> tareas) {
		this.tareas = tareas;
	}
	
	public ArrayList<Tarea> getTareas() { 
		return this.tareas;
	}
	
	public void agregarTarea(Tarea tarea) {
		this.tareas.add(tarea);
	}
}

File: python/calendario/dia.py
from python.revErrores.revErrores import revErrores
from tareas.tarea import *
from persistencia.serializacion import Serializador

#Esta clase seran los dias que haran parte de cada planificador y contendra las tareas a realizar o para entregar dicho dia
class Dia():

    def __init__(self, etiqueta, fecha, tareas = []):
        revErrores.formatoFecha(fecha)
        self._etiqueta = etiqueta
        self._fecha = fecha
        self._tareas = tareas
        Serializador.dias.append(self)
        Serializador.guardarDias(Serializador.dias)
    
    def mostrarTarea(self): #muestra las tareas almacenadas en el dia
        Serializador.diasGuardados()
        Serializador.tareasGuardadas()

        l = ""
        for t in self._tareas:
            l += "---" + t.GetTitulo() + "\n"
        return l

    def agragarTarea(self, tarea): #toma una tarea y si la echa coincide con el dia la agrega
        if tarea.fechaDeEntrega == self._fecha:
            self._tareas.append(tarea)
        else:
            return 'La fecha de entrega de esta tarea no coincide con el dia' 

    def setEtiqueta(self, eti):
        self._etiqueta = eti

    def getEtiqueta(self):
        return self._etiqueta

    def setFecha(self, fecha):
        revErrores.formatoFecha
        self._fecha = fecha

    def getFecha(self):
        return self._fecha
    
    def setTareas(self, tareas):
        self._tareas = tareas

    def getTareas(self):
        return self._tareas